packages/manager/apps/ips/src/data/hooks/useServiceList.ts

```diff
@@ -0,0 +1,58 @@
+import { useQueries } from '@tanstack/react-query';
+import { getVrackList } from '../api/vrack';
+import { ServiceType } from '../api/ips';
+
+export const ipParkingOptionValue = 'parking';
+
+/**
+ * Fetch the list of available services to order an additional IP
+ */
+export const useServiceList = () => {
+  const queries = useQueries({
+    queries: [
+      {
+        queryKey: ['dedicatedCloud'],
+        queryFn: () => [],
+      },
+      {
+        queryKey: ['server'],
+        queryFn: () => [],
+      },
+      {
+        queryKey: ['vps'],
+        queryFn: () => [],
+      },
+      {
+        queryKey: ['vrack'],
+        queryFn: getVrackList,
+      },
+    ],
+  });
+
+  return {
+    getServiceType: (serviceId: string) => {
+      if (serviceId === ipParkingOptionValue) {
+        return ServiceType.ipParking;
+      }
+      if (queries[0]?.data?.includes(serviceId)) {
+        return ServiceType.dedicatedCloud;
+      }
+      if (queries[1]?.data?.includes(serviceId)) {
+        return ServiceType.server;
+      }
+      if (queries[2]?.data?.includes(serviceId)) {
+        return ServiceType.vps;
+      }
+      return queries[3]?.data?.data?.includes(serviceId)
+        ? ServiceType.vrack
+        : ServiceType.unknown;
+    },
```

Comment:
Possible to use switch here ?

Reply:
I don't think we can do this with the `switch` syntax
It's made to test a variable against multiple values (like the first case) but the other cases test the inclusion inside an array of values